src/components/Portal/PortalInner.tsx

```diff
@@ -0,0 +1,48 @@
+import * as PropTypes from 'prop-types'
+import { invoke } from 'lodash'
+import { Component } from 'react'
+import { createPortal } from 'react-dom'
+import { Extendable } from '../../../types/utils'
+
+export interface IPortalProps {
+  onMount?: (props: IPortalProps) => void
+  onUnmount?: (props: IPortalProps) => void
+}
+
+/**
+ * An inner component that allows you to render children outside their parent.
+ */
+class PortalInner extends Component<Extendable<IPortalProps>, any> {
+  public static propTypes = {
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /**
+     * Called when the portal is mounted on the DOM
+     *
+     * @param {object} data - All props.
+     */
+    onMount: PropTypes.func,
+
+    /**
+     * Called when the portal is unmounted from the DOM
+     *
+     * @param {object} data - All props.
+     */
+    onUnmount: PropTypes.func,
+  }
+
+  public componentDidMount() {
+    invoke(this.props, 'onMount', { ...this.props })
+  }
+
+  public componentWillUnmount() {
+    invoke(this.props, 'onUnmount', { ...this.props })
+  }
+
+  public render() {
+    return createPortal(this.props.children, document.body)
```

Comment:
Accessing `document` will fail in server side rendering.  This should probably be a `defaultProp` instead that is safe guarded by a browser check first:

```jsx
import { isBrowser } from '../../lib'

static defaultProps = {
  context: isBrowser() ? document.body : null
}

render() {
  const { children, context } = this.props

  return createPortal(children, context)
}
```

Reply:
I made the change but I don't really understand it. Server side rendering? Where do we do that?